### PR TITLE
fixed regression where auth no longer works in publishing subnet

### PR DESCRIPTION
**Pull request by allandegnan**
### What

Publishing subnets require auth.  Switching to the SDK broke the auth, this should hopefully restore it.

### How to review

Requires https://github.com/ONSdigital/dp-search-api/pull/184 

### Who can review

!me

### CODE REVIEW

File: handlers/handlers.go [hunk 1]
```diff
@@ -183,6 +186,9 @@ func getCategoriesCountQuery(searchQuery url.Values) url.Values {
 func getCategoriesTypesCount(ctx context.Context, accessToken, collectionID string, query url.Values, searchC SearchClient, censusTopicCache *cache.Topic) ([]data.Category, []data.Topic, error) {
 	var options searchSDK.Options
 	options.Query = query
+	options.Headers = map[searchSDK.Header][]string{
+		searchSDK.Authorisation: {"Bearer " + accessToken},
```
Comment (nshumoogum) [inline handlers/handlers.go:190]: We need the collection id too.

Do we need to extend the list of available headers in the SDK package to include this? Or can we create our on key value pairs?

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: handlers/handlers.go (hunk 1 above)
Comment (allandegnan) [reply handlers/handlers.go:190]: You don't need to export the actual headers, just the Header struct.  You can write code like this:

``` 
options.Headers = map[searchSDK.Header][]string{
		searchSDK.Authorisation: {"Bearer " + accessToken},
		"header_name": { "header_value" },
```

The list of headers is just a nice to have.

Comment (nshumoogum) [reply handlers/handlers.go:191]: Yep, I thought as much. So we just need to figure out what the header is that holds the collection id and then add that to the map of headers

Comment (allandegnan) [reply handlers/handlers.go:191]: Yeah I've updated both PRs now.

https://github.com/ONSdigital/dp-search-api/pull/184

**Review by nshumoogum — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -30,7 +30,7 @@ require (
 	github.com/ONSdigital/dp-elasticsearch/v3 v3.0.0-alpha.7 // indirect
 	github.com/ONSdigital/dp-mongodb-in-memory v1.3.1 // indirect
 	github.com/ONSdigital/dp-rchttp v1.0.0 // indirect
-	github.com/ONSdigital/dp-search-api v1.29.0 // indirect
+	github.com/ONSdigital/dp-search-api v1.7.1-0.20230120122810-91b330fb217d // indirect
```
Comment (nshumoogum) [inline go.mod:33]: We look to only add dependencies of our tagged releases

So will need to get dp-search-api released and tagged and then amend the go.mod file to point to new version here

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (allandegnan) [reply go.mod]: Fixed now

**Review by nshumoogum — APPROVED** [review]

**Pull request merged**